vowpalwabbit/core/src/parser.cc

```diff
@@ -622,7 +626,14 @@ void enable_sources(VW::workspace& all, bool quiet, size_t passes, input_options
         all.example_parser->reader = VW::parsers::flatbuffer::flatbuffer_to_examples;
       }
 #endif
-
+#ifdef BUILD_CSV
+      else if (input_options.csv_opts && input_options.csv_opts->enabled)
+      {
+        auto csv_parser = VW::make_unique<VW::parsers::csv::parser>(*(input_options.csv_opts.get()), all.logger);
```

Comment:
You can just do
`all.custom_parser = VW::make_unique<VW::parsers::csv::parser>(*(input_options.csv_opts.get()), all.logger);`

It will do the polymorphism for you

Reply:
Cool, thanks!